### DISCUSSION TITLE
Seeding with environmental variable fails

### DISCUSSION THREAD

**Discussion by seed-of-apricot** [prisma/prisma#2774] [Q&A] [answered]
When I run the following code for seeding:

```ts
import { PrismaClient } from '@prisma/client';
import elements from './data/elements.json';

const prisma = new PrismaClient();

async function seed() {
  elements.map(async (element) => {
    await prisma.element.create({
      data: element,
    });
  });
}

seed()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    console.log('seeding completed');
    await prisma.disconnect();
  });
```

It fails with

```node
(node:11640) UnhandledPromiseRejectionWarning: Error: error: Environment variable not found: DATABASE_URL.
  -->  schema.prisma:3
   |
 2 |   provider = "postgresql"
 3 |   url      = env("DATABASE_URL")
```

despite the environmental variable `DATABASE_URL` in the prisma folder. When I run `npx prisma generate` it correctly recognize the env file. How can I let it understand the env file when seeding?

**Comment by janpio** [comment]
What version of Prisma are you using?
Did this work before?

**Reply by seed-of-apricot** [reply]
I use 2.0.0, the first GA version. I didn't try seeding from the cli before.

**Reply by ryands17** [reply]
@seed-of-apricot How are you passing the environment variables? Are you using something like [dotenv-cli](https://www.npmjs.com/package/dotenv-cli)

If not, Prisma wouldn't be able to find the environment variable.

**Reply by janpio** [reply]
Are you sure about that @ryands17? Shouldn't `prisma/.env` be picked up automatically? Is that where you put it @seed-of-apricot?

**Reply by seed-of-apricot** [reply, marked as answer]
I use prisma/.env as the tutorial says and importantly it works when I run `npx prisma migrate` or `generate`.

**Comment by ludralph** [comment]
Hi @seed-of-apricot,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
